### PR TITLE
ENH: linalg.funm: Pythranize double for loop

### LINKED ISSUE #4708: ENH: linalg.funm: a possible speed-up
`scipy.linalg.funm` contains a double loop in pure python: https://github.com/scipy/scipy/blob/master/scipy/linalg/matfuncs.py#L609

The whole algorithm (9.1.1. in the 4th edition of Golub & van Loan) is likely not easily vectorizable, but at least it might be worth investigating if the looping can be moved to cython.

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue
Closes gh-4708

#### What does this implement/fix?
gh-4708 noted that there is a pure Python double `for` loop in `scipy.linalg.funm` and suggested Cythonizing it. This PR  tries Pythranizing it, instead. 

#### Additional information 
Example:
```python3
import numpy as np
from scipy.linalg import funm
rng = np.random.default_rng(123456)
A = rng.random((100, 100))
%timeit funm(A, lambda x: x*x)
# pythran: 11.7 ms ± 1.97 ms per loop (mean ± std. dev. of 7 runs, 100 loops each)
# main: 50.1 ms ± 12.2 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

### CODE REVIEW

File: scipy/linalg/_linalg_pythran.py
Comment (mdhaber) [inline scipy/linalg/_linalg_pythran.py:11]: This works to resolve the lint failure. So that it was more obvious that this was just copy-pasted from `_matfuncs.py`, I did:
```suggestion
            ks = slice(i, j-1)
            val = (sum(T[i - 1, ks] * F[ks, j - 1]) - sum(F[i - 1, ks] * T[ks, j - 1]))
```
But personally I'm happy with what you have here; I just wanted to make it obvious that the code is the same except `dot` is replaced with `*` and `sum`.

**Review by mdhaber — APPROVED** [review]
I don't know why it wasn't working for me. Your branch builds for me : )

Approving but I'll let Evgeni take a look since he filed the issue.

*Update: CI failure does not look related.*

File: scipy/linalg/_matfuncs.py
Comment (ev-br) [inline scipy/linalg/_matfuncs.py:688]: Mind want to duplicate the comment in the `_linalg_pythran.py` file.

**Review by ev-br — APPROVED** [review]
Looks reasonable!

Looks like test coverage of `funm` is rather thin, but improving it does not need to block this PR. If there's a follow-upm it might be helpful to add a comment to the pythran source with an algorithm reference. But it's one grep away at  `_funm_loops` call site, so probably even OK as is. 

None of these two nitpicks need to block this PR, so merging. Thanks Jake, Matt!

**Pull request merged**
